### PR TITLE
Add STL to Assemblys export formats

### PR DESCRIPTION
**Pull request by qwelyt**
Turn the assembly into a compound and use Shapes export method to export
the assembly. This way the export function does not need to be
duplicated, and it is what users have been doing manually to get their
objects.

Should solve https://github.com/CadQuery/cadquery/issues/965

### CODE REVIEW

**Comment by SDI8** [comment]
Shouldn't all of the "shape-formats" be supported then (and not just STL)?
This situation is very similar to what is happening with Workplanes in [`exporters.export`](https://github.com/CadQuery/cadquery/blob/a5fadebe72b2a46881c3847cca748b8cf716ccd0/cadquery/occ_impl/exporters/__init__.py#L59), where any Workplane gets converted to a compound first. 
You could achieve similar functionality to your proposal by allowing Assemblies in[ `exporters.export`](https://github.com/CadQuery/cadquery/blob/a5fadebe72b2a46881c3847cca748b8cf716ccd0/cadquery/occ_impl/exporters/__init__.py#L36) - although I'm not in a position to foresee all issues that that could entail.

**Comment by jmwright** [comment]
@lorenzncode @adam-urbanczyk Any thoughts on this?

**Comment by adam-urbanczyk** [comment]
STL has been discussed already. It does not support all the features of assy (color), but OK popular request and whatnot. -1 for other formats though.

**Comment by lorenzncode** [comment]
I wasn't sure if the discussion in #965 had concluded.  Regarding the use of such an STL, it may be that the user wants to access the solver functionality for a particular part.  A user on Discord mentioned this approach, and asked "Am I misusing Assembly?".

**Review by adam-urbanczyk — APPROVED** [review]

File: cadquery/assembly.py [hunk 1]
```diff
@@ -468,6 +468,8 @@ def save(
             exportGLTF(self, path, True, tolerance, angularTolerance)
         elif exportType == "VTKJS":
             exportVTKJS(self, path)
+        elif exportType == "STL":
+            self.toCompound().exportStl(path, tolerance, angularTolerance)
```
Comment (lorenzncode) [inline cadquery/assembly.py:472]: Add support for ascii?  ` assy.save("out.stl", ascii=True)`

```py 
            ascii = False
            if "ascii" in kwargs and kwargs["ascii"]:
                ascii = True
            self.toCompound().exportStl(path, tolerance, angularTolerance, ascii)
```

Comment (lorenzncode) [reply cadquery/assembly.py:472]: ascii support could be revisited in a separate PR.

**Review by lorenzncode — APPROVED** [review]

**Comment by adam-urbanczyk** [comment]
Should I merge @jmwright @lorenzncode ?

**Comment by jmwright** [comment]
+1

**Comment by lorenzncode** [comment]
+1

**Comment by adam-urbanczyk** [comment]
Thanks @qwelyt !

**Pull request merged**